src/awkward/_nplikes/typetracer.py

```diff
@@ -506,6 +506,8 @@ def _apply_ufunc(self, ufunc, *inputs):
             assert not isinstance(x, PlaceholderArray)
             try_touch_data(x)
 
+        inputs = [x.content if isinstance(x, MaybeNone) else x for x in inputs]
+
```

Comment:
In general, we want `MaybeNone` to erase in contexts where a `None` would just error. We probably need to do more here, but this seems like a good start.

This was motivated by a failing test.